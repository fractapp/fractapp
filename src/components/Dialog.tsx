import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {BlueButton} from 'components/BlueButton';
import PopDialog, {DialogContent} from 'react-native-popup-dialog';

/**
 * Popup dialog component with 1 button
 * @category Components
 */
export const Dialog = ({
  title,
  text,
  visible,
  onPress,
}: {
  title: string;
  text: string;
  visible: boolean;
  onPress: () => void;
}) => {
  return (
    <PopDialog
      visible={visible}
      width={0.7}
      dialogTitle={<Text style={styles.title}>{title}</Text>}
      footer={
        <View style={styles.btnImg}>
          <BlueButton text={'OK'} height={40} onPress={onPress} />
        </View>
      }>
      <DialogContent style={styles.dialog}>
        <Text style={styles.description}>{text}</Text>
      </DialogContent>
    </PopDialog>
  );
};

const styles = StyleSheet.create({
  dialog: {
    alignItems: 'center',
    padding: 0,
  },
  title: {
    marginTop: 10,
    textAlign: 'center',
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    color: 'black',
  },
  description: {
    marginTop: 5,
    width: '100%',
    textAlign: 'center',
    fontSize: 14,
    fontFamily: 'Roboto-Regular',
    color: '#888888',
  },
  btnImg: {
    marginBottom: 12,
    width: '40%',
    alignSelf: 'center',
  },
});
