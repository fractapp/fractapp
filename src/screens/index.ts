export {ImportWallet} from './ImportWallet';
export {SaveSeed} from './SaveSeed';
export {SaveWallet} from './SaveWallet';
export {SettingWallet} from './SettingWallet';
export {Start} from './Start';
export {WalletDetails} from './WalletDetails';
export {WalletFileBackup} from './WalletFileBackup';
export {Wallets} from './Wallets';
export {Settings} from './Settings';
export {ConfirmSaveSeed} from './ConfirmSaveSeed';
export {ImportSeed} from './ImportSeed';
export {WalletFileImport} from './WalletFileImport';
export {GoogleDrivePicker} from './GoogleDrivePicker';
export {Receive} from './Receive';
export {Milestone} from './Milestone';
export {TransactionDetails} from './TransactionDetails';
export {SplashScreen} from './SplashScreen';
export {Backup} from './Backup';
export {NewPassCode} from './NewPassCode';
export {VerifyPassCode} from './VerifyPassCode';
export {EditProfile} from './EditProfile';
export {EditPhoneNumber} from './EditPhoneNumber';
export {SelectCountry} from './SelectCountry';
export {ConfirmCode} from './ConfirmCode';
export {EditEmail} from './EditEmail';
export {Navigation} from './Navigation';
export {Chats} from './Chats';
export {Chat} from './Chat';
export {Connecting} from './Connecting';
export {EditUsername} from './EditUsername';
export {EditName} from './EditName';
export {Search} from './Search';
export {SelectWallet} from './SelectWallet';
export {Send} from './Send';
export {EnterAmount} from './EnterAmount';
export {SendWithEnterAddress} from './SendWithEnterAddress';
