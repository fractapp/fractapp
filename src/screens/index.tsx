export { default as ImportWallet } from './ImportWallet';
export { default as SaveSeed } from './SaveSeed';

export { default as SaveWallet } from './SaveWallet';
export { default as SettingWallet } from './SettingWallet';
export { default as Start } from './Start';

export { default as WalletDetails } from './WalletDetails';

export { default as WalletEncryption } from './WalletEncryption';
export { default as Wallets } from './Wallets';
export { ConfirmSaveSeed } from './ConfirmSaveSeed'