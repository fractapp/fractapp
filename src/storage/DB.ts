import {Account} from 'models/account';
import AsyncStorage from '@react-native-async-storage/async-storage';
import Keychain from 'react-native-keychain';
import {base64Encode, randomAsU8a} from '@polkadot/util-crypto';
import PasscodeUtil from 'utils/passcode';
import {Keyring} from '@polkadot/keyring';
import {u8aToHex} from '@polkadot/util';
import {Currency, getSymbol} from 'models/wallet';
import {Transaction} from 'models/transaction';
import {ChatInfo} from 'models/chatInfo';
import {AuthInfo} from 'models/authInfo';
import {MyProfile} from 'models/myProfile';
import BN from 'bn.js';
import {UserProfile} from 'models/profile';
import {Use} from 'react-native-svg';
/**
 * @namespace
 * @category storage
 */

namespace DB {
  const secureOption = {
    service: 'com.fractapp',
  };
  const biometryOption = {
    service: 'com.fractapp',
    accessControl: Keychain.ACCESS_CONTROL.BIOMETRY_CURRENT_SET,
  };

  const AsyncStorageKeys = {
    price: (currency: Currency) => `price_${currency}`,
    profile: 'profile',
    authInfo: 'auth_info',
    notificationCount: 'notification_count',
    accounts: 'accounts',
    contacts: 'contacts',
    users: 'users',
    accountInfo: (address: string) => `account_${address}`,
    transactions: (currency: Currency) => `transactions_${getSymbol(currency)}`,
    pendingTransactions: (currency: Currency) =>
      `pending_transactions_${getSymbol(currency)}`,
    chatByAddress: (address: string) => `chat_${address}`,
    chatsInfo: 'chats_info',
  };
  const SecureStorageKeys = {
    firebaseToken: 'firebase_token',
    authJWT: 'auth_jwt',
    seed: 'seed',
    salt: 'salt',
    passcodeHash: 'passcode_hash',
  };
  const PasscodeStorageKeys = {
    passcode: 'passcode',
  };

  export async function setSecureItem(key: string, value: string) {
    const result = await Keychain.setInternetCredentials(
      key,
      key,
      value,
      secureOption,
    );
    if (result == false) {
      throw `invalid set ${key}`;
    }
  }
  export async function getSecureItem(key: string): Promise<string | null> {
    const result = await Keychain.getInternetCredentials(key, secureOption);
    if (result === false) {
      return null;
    }

    return result.password;
  }

  export async function setAuthInfo(auth: AuthInfo) {
    await AsyncStorage.setItem(AsyncStorageKeys.authInfo, JSON.stringify(auth));
  }
  export async function getAuthInfo(): Promise<AuthInfo | null> {
    const result = await AsyncStorage.getItem(AsyncStorageKeys.authInfo);
    if (result == null) {
      return null;
    }
    return JSON.parse(result);
  }

  export async function setPrice(currency: Currency, value: number) {
    await AsyncStorage.setItem(AsyncStorageKeys.price(currency), String(value));
  }
  export async function getPrice(currency: Currency): Promise<number> {
    const result = await AsyncStorage.getItem(AsyncStorageKeys.price(currency));
    if (result == null) {
      return 0;
    }
    return Number(result);
  }

  export async function setProfile(profile: MyProfile) {
    await AsyncStorage.setItem(
      AsyncStorageKeys.profile,
      JSON.stringify(profile),
    );
  }
  export async function getProfile(): Promise<MyProfile | null> {
    const result = await AsyncStorage.getItem(AsyncStorageKeys.profile);
    if (result == null) {
      return null;
    }
    return JSON.parse(result);
  }

  export async function setNotificationCount(count: number) {
    await AsyncStorage.setItem(
      AsyncStorageKeys.notificationCount,
      String(count),
    );
  }
  export async function getNotificationCount(): Promise<number> {
    const result = await AsyncStorage.getItem(
      AsyncStorageKeys.notificationCount,
    );
    return Number(result);
  }

  export async function enablePasscode(passcode: string, isBiometry: boolean) {
    if (isBiometry) {
      const result = await Keychain.setInternetCredentials(
        PasscodeStorageKeys.passcode,
        PasscodeStorageKeys.passcode,
        passcode,
        biometryOption,
      );
      if (result === false) {
        throw 'invalid set passcode';
      }
    }

    const salt = base64Encode(randomAsU8a(32));
    const hash = PasscodeUtil.hash(passcode, salt);
    let authInfo = await getAuthInfo();
    if (authInfo == null) {
      authInfo = new AuthInfo(false, false, false, false);
    }

    try {
      await setSecureItem(SecureStorageKeys.salt, salt);
      await setSecureItem(SecureStorageKeys.passcodeHash, hash);
      authInfo.isPasscode = true;
      authInfo.isBiometry = isBiometry;
    } catch (e) {
      await disablePasscode();
    }

    await setAuthInfo(authInfo);
  }
  export async function disablePasscode() {
    let authInfo = await getAuthInfo();
    if (authInfo == null) {
      authInfo = new AuthInfo(false, false, false, false);
    }
    authInfo.isPasscode = false;
    authInfo.isBiometry = false;

    await setAuthInfo(authInfo);

    await Keychain.resetInternetCredentials(
      SecureStorageKeys.salt,
      secureOption,
    );
    await Keychain.resetInternetCredentials(
      SecureStorageKeys.passcodeHash,
      secureOption,
    );
  }

  export async function getSalt(): Promise<string | null> {
    return await getSecureItem(SecureStorageKeys.salt);
  }
  export async function getPasscodeHash(): Promise<string | null> {
    return await getSecureItem(SecureStorageKeys.passcodeHash);
  }
  export async function getPasscode(): Promise<string> {
    const result = await Keychain.getInternetCredentials(
      PasscodeStorageKeys.passcode,
      biometryOption,
    );
    if (result === false) {
      throw 'value by passcode key not found';
    }

    return result.password;
  }

  export async function createAccounts(seed: string) {
    let polkadotWallet = new Keyring({
      type: 'sr25519',
      ss58Format: 0,
    }).addFromUri(seed);
    let kusamaWallet = new Keyring({type: 'sr25519', ss58Format: 2}).addFromUri(
      seed,
    );
    let accountsInfo = new Array<Account>(
      new Account(
        'Polkadot wallet',
        polkadotWallet.address,
        u8aToHex(polkadotWallet.publicKey),
        Currency.Polkadot,
        0,
        new BN(0).toString(10),
      ),
      new Account(
        'Kusama wallet',
        kusamaWallet.address,
        u8aToHex(kusamaWallet.publicKey),
        Currency.Kusama,
        0,
        new BN(0).toString(10),
      ),
    );
    let accounts = new Array<string>();
    for (let i = 0; i < accountsInfo.length; i++) {
      const account = accountsInfo[i];
      accounts.push(account.address);
      await setAccountInfo(account);
    }
    await setAccounts(accounts);
    await setSecureItem(SecureStorageKeys.seed, seed);
  }
  export async function getSeed(): Promise<string | null> {
    return await getSecureItem(SecureStorageKeys.seed);
  }

  export async function setAccounts(accounts: Array<string>) {
    await AsyncStorage.setItem(
      AsyncStorageKeys.accounts,
      JSON.stringify(accounts),
    );
  }
  export async function getAccounts(): Promise<Array<string> | null> {
    const result = await AsyncStorage.getItem(AsyncStorageKeys.accounts);
    if (result == null) {
      return null;
    }
    return JSON.parse(result);
  }

  export async function setChatsInfo(chats: Map<string, ChatInfo>) {
    await AsyncStorage.setItem(
      AsyncStorageKeys.chatsInfo,
      JSON.stringify([...chats]),
    );
  }
  export async function getChatsInfo(): Promise<Map<string, ChatInfo>> {
    const result = await AsyncStorage.getItem(AsyncStorageKeys.chatsInfo);

    if (result == null) {
      return new Map<string, ChatInfo>();
    }
    return new Map<string, ChatInfo>(JSON.parse(result));
  }

  export async function setChat(address: string, chats: Map<string, Currency>) {
    await AsyncStorage.setItem(
      AsyncStorageKeys.chatByAddress(address),
      JSON.stringify([...chats]),
    );
  }
  export async function getChat(
    address: string,
  ): Promise<Map<string, boolean>> {
    const result = await AsyncStorage.getItem(
      AsyncStorageKeys.chatByAddress(address),
    );

    if (result == null) {
      return new Map<string, boolean>();
    }
    return new Map<string, boolean>(JSON.parse(result));
  }

  export async function setTx(currency: Currency, tx: Transaction) {
    const txs = await getTxs(currency);

    txs.set(tx.id, tx);
    await AsyncStorage.setItem(
      AsyncStorageKeys.transactions(currency),
      JSON.stringify([...txs]),
    );
  }
  export async function getTxs(
    currency: Currency,
  ): Promise<Map<string, Transaction>> {
    const result = await AsyncStorage.getItem(
      AsyncStorageKeys.transactions(currency),
    );

    if (result == null) {
      return new Map<string, Transaction>();
    }
    return new Map<string, Transaction>(JSON.parse(result));
  }

  export async function setContacts(contacts: Array<UserProfile>) {
    await AsyncStorage.setItem(
      AsyncStorageKeys.contacts,
      JSON.stringify(contacts),
    );
  }

  export async function getContacts(): Promise<Array<UserProfile>> {
    const result = await AsyncStorage.getItem(AsyncStorageKeys.contacts);

    if (result == null) {
      return [];
    }

    return JSON.parse(result);
  }

  export async function setUsers(users: Map<string, UserProfile>) {
    await AsyncStorage.setItem(
      AsyncStorageKeys.users,
      JSON.stringify([...users]),
    );
  }

  export async function getUsers(): Promise<Map<string, UserProfile>> {
    const result = await AsyncStorage.getItem(AsyncStorageKeys.users);

    if (result == null) {
      return new Map<string, UserProfile>();
    }
    return new Map<string, UserProfile>(JSON.parse(result));
  }

  export async function setPendingTxs(currency: Currency, txs: Array<string>) {
    await AsyncStorage.setItem(
      AsyncStorageKeys.pendingTransactions(currency),
      JSON.stringify(txs),
    );
  }

  export async function getPendingTxs(
    currency: Currency,
  ): Promise<Array<string>> {
    const result = await AsyncStorage.getItem(
      AsyncStorageKeys.pendingTransactions(currency),
    );

    if (result == null) {
      return [];
    }

    return JSON.parse(result);
  }

  export async function setAccountInfo(account: Account) {
    await AsyncStorage.setItem(
      AsyncStorageKeys.accountInfo(account.address),
      JSON.stringify(account),
    );
  }
  export async function getAccountInfo(
    address: string,
  ): Promise<Account | null> {
    const result = await AsyncStorage.getItem(
      AsyncStorageKeys.accountInfo(address),
    );
    if (result == null) {
      return null;
    }
    return JSON.parse(result);
  }

  export async function setFirebaseToken(token: string) {
    await setSecureItem(SecureStorageKeys.firebaseToken, token);
  }
  export async function getFirebaseToken(): Promise<string | null> {
    return await getSecureItem(SecureStorageKeys.firebaseToken);
  }

  export async function setJWT(token: string) {
    await setSecureItem(SecureStorageKeys.authJWT, token);
  }
  export async function getJWT(): Promise<string | null> {
    return await getSecureItem(SecureStorageKeys.authJWT);
  }
}

export default DB;
