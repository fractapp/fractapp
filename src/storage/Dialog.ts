import {createContext, Dispatch} from 'react';

/**
 * @namespace
 * @category Context storage
 */
namespace DialogStore {
  export enum Action {
    OPEN,
    CLOSE,
  }

  type State = {
    title: string;
    text: string;
    visible: boolean;
    onPress?: () => void;
  };

  export const initialState = (): State => ({
    text: '',
    title: '',
    visible: false,
  });

  export type ContextType = {
    state: State;
    dispatch: Dispatch<any>;
  };

  export const Context = createContext<ContextType>({
    state: initialState(),
    dispatch: () => null,
  });

  export function reducer(prevState: State, action: any): State {
    switch (action.type) {
      case Action.OPEN:
        return {
          title: action.title,
          text: action.text,
          onPress: action.onPress,
          visible: true,
        };
      case Action.CLOSE:
        let newDialog = Object.assign({}, prevState);
        newDialog.visible = false;
        return newDialog;
      default:
        return prevState;
    }
  }

  export const open = (title: string, text: string, onPress: () => void) => ({
    type: Action.OPEN,
    title: title,
    text: text,
    onPress: onPress,
  });
  export const close = () => ({
    type: Action.CLOSE,
  });
}
export default DialogStore;
