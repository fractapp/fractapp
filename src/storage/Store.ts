import { configureStore } from '@reduxjs/toolkit';
import ChatsStore from 'storage/Chats';
import DialogStore from 'storage/Dialog';
import GlobalStore from 'storage/Global';
import ServerInfoStore from 'storage/ServerInfo';
import AccountsStore from 'storage/Accounts';
import UsersStore from 'storage/Users';

/**
 * @namespace
 * @category Storage
 */
namespace Store {
  export const initStore = () => {
    return configureStore({
      reducer: {
        global: GlobalStore.reducer,
        accounts: AccountsStore.reducer,
        chats: ChatsStore.reducer,
        dialog: DialogStore.reducer,
        serverInfo: ServerInfoStore.reducer,
        users: UsersStore.reducer,
      },
    });
  };
  export const initValues = () => ({
    global: GlobalStore.initialState(),
    accounts: AccountsStore.initialState(),
    chats: ChatsStore.initialState(),
    dialog: DialogStore.initialState(),
    serverInfo: ServerInfoStore.initialState(),
    users: UsersStore.initialState(),
  });

  export type States = {
    global: GlobalStore.State,
    accounts: AccountsStore.State,
    chats: ChatsStore.State,
    dialog: DialogStore.State,
    serverInfo: ServerInfoStore.State,
    users: UsersStore.State,
  }
}
export default Store;
