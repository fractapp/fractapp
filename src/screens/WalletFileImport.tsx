import React, {useState, useEffect} from 'react';
import {StyleSheet, View, Text, NativeModules} from 'react-native';
import {BlueButton} from 'components/BlueButton';
import {PasswordInput} from 'components/PasswordInput';
import backupUtil from 'utils/backup';
import {FileBackup} from 'types/backup';
import GlobalStore from 'storage/Global';
import Dialog from 'storage/Dialog';
import StringUtils from 'utils/string';
import { useDispatch, useSelector } from 'react-redux';
import tasks from 'utils/tasks';

/**
 * Wallet file import screen
 * @category Screens
 */
export const WalletFileImport = ({route}: {route: any}) => {
  const dispatch = useDispatch();
  const globalState: GlobalStore.State = useSelector((state: any) => state.global);

  const [password, setPassword] = useState<string>('');

  const file: FileBackup = route.params.file;

  const startImport = async () => {
    dispatch(GlobalStore.actions.showLoading());
  };

  useEffect(() => {
    dispatch(GlobalStore.actions.showLoading());

    NativeModules.PreventScreenshotModule.forbid().then((result: string) =>
      console.log(result),
    );

    return () =>
      NativeModules.PreventScreenshotModule.allow().then((result: string) =>
        console.log(result),
      );
  }, []);

  useEffect(() => {
    if (!globalState.loadInfo.isLoadingShow) {
      return;
    }

    (async () => {
      let seed = '';
      try {
        seed = await backupUtil.getSeed(file, password);
      } catch (e) {
        dispatch(
          Dialog.actions.showDialog({
              title: StringUtils.texts.walletFileImport.invalidPasswordTitle,
              text: '',
            }
          ),
        );
        dispatch(GlobalStore.actions.hideLoading());
        return;
      }

      await tasks.createAccount(seed, dispatch);
      dispatch(GlobalStore.actions.hideLoading());
    })();
  }, [globalState.loadInfo.isLoadingShow]);

  return (
    <View
      style={{
        flexDirection: 'column',
        flex: 1,
        alignItems: 'center',
      }}>
      <Text style={styles.title}>
        {StringUtils.texts.walletFileImport.title}
      </Text>
      <Text style={styles.description}>
        {StringUtils.texts.walletFileImport.description}
      </Text>

      <View style={styles.newPassword}>
        <PasswordInput
          onChangeText={(value: string) => setPassword(value)}
          placeholder={StringUtils.texts.walletFileImport.passwordPlaceholder}
        />
      </View>

      <View style={{width: '80%', position: 'absolute', bottom: 40}}>
        <BlueButton
          text={StringUtils.texts.RestoreBtn}
          height={50}
          onPress={startImport}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  title: {
    marginTop: 80,
    fontSize: 25,
    fontFamily: 'Roboto-Regular',
    color: '#2AB2E2',
  },
  description: {
    textAlign: 'center',
    width: '90%',
    marginTop: 40,
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    color: '#888888',
  },
  newPassword: {
    marginTop: 30,
    width: '90%',
  },
  confirmPassword: {
    marginTop: 20,
    width: '90%',
  },
  error: {
    marginTop: 20,
    color: 'red',
    fontFamily: 'Roboto-Regular',
    fontSize: 15,
  },
});
