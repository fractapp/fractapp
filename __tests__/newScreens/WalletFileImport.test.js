import React, {useContext, useState} from 'react';
import {WalletFileImport} from 'screens/WalletFileImport';
import renderer from 'react-test-renderer';
import GlobalStore from 'storage/Global';
import {fireEvent, render} from '@testing-library/react-native';
import BackupUtils from 'utils/backup';
import DB from 'storage/DB';
import StringUtils from 'utils/string';
import { NativeModules } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import Store from 'storage/Store';
import tasks from 'utils/tasks';
import Dialog from 'storage/Dialog';

jest.mock('react-native-crypto', () => {});
jest.mock('storage/DB', () => ({}));
jest.mock('utils/tasks', () => ({
  createAccount: jest.fn(),
}));
jest.mock('utils/backup', () => ({
  getSeed: jest.fn(),
}));
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));
jest.mock('react-native-i18n', () => ({
  t: (value) => value,
}));
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

NativeModules.PreventScreenshotModule = {
  forbid: async () => {},
  allow: async () => {},
};
jest.useFakeTimers();

useState.mockImplementation((init) => [init, jest.fn()]);

const dispatch = jest.fn();
useDispatch.mockReturnValue(dispatch);

let store = Store.initValues();
useSelector.mockImplementation((fn) => {
  return fn(store);
});

it('Test positive', () => {
  const tree = renderer
    .create(<WalletFileImport route={{params: {file: {}}}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test loading', () => {
  useState
    .mockImplementationOnce((init) => [init, jest.fn()])
    .mockImplementationOnce(() => [true, jest.fn()]);

  const tree = renderer
    .create(<WalletFileImport route={{params: {file: {}}}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test useEffect', async () => {
  const setLoading = jest.fn();
  useState
    .mockImplementationOnce((init) => ['123123', jest.fn()])
    .mockImplementationOnce(() => [true, setLoading]);

  const file = {
    seed: 'seed',
    algorithm: 'algorithm',
  };

  BackupUtils.getSeed.mockReturnValueOnce(file.seed);

  await render(
    <WalletFileImport route={{params: {file: file}}} />,
  );

  expect(BackupUtils.getSeed).toBeCalledWith(file, '123123');
  expect(tasks.createAccount).toBeCalledWith('seed', dispatch);
});

it('Test useEffect throw', async () => {
  const setLoading = jest.fn((f) => f);
  useState
    .mockImplementationOnce((init) => ['123123', jest.fn()])
    .mockImplementationOnce(() => [true, setLoading(false)]);

  const file = {
    seed: 'seed',
    algorithm: 'algorithm',
  };

  BackupUtils.getSeed.mockImplementationOnce(() => {
    throw new Error('error');
  });

  await render(
    <WalletFileImport route={{ params: { file: file } }} />,
  );

  expect(dispatch).toBeCalledWith(
    Dialog.actions.showDialog({
        title: StringUtils.texts.walletFileImport.invalidPasswordTitle,
        text: '',
      }
    )
  );
  expect(setLoading).toBeCalledWith(false);
});

it('Test click Decrypt', () => {
  const setLoading = jest.fn();
  useState
    .mockImplementationOnce((init) => [init, jest.fn()])
    .mockImplementationOnce(() => [false, setLoading]);

  const component = render(<WalletFileImport route={{params: {file: {}}}} />);
  fireEvent.press(component.getByText(StringUtils.texts.RestoreBtn));

  expect(setLoading).toBeCalledWith(true);
});
