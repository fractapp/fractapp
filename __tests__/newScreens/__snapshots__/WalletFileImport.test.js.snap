// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test loading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2AB2E2",
        "fontFamily": "Roboto-Regular",
        "fontSize": 25,
        "marginTop": 80,
      }
    }
  >
    wallet_file_import.title
  </Text>
  <Text
    style={
      Object {
        "color": "#888888",
        "fontFamily": "Roboto-Regular",
        "fontSize": 15,
        "marginTop": 40,
        "textAlign": "center",
        "width": "90%",
      }
    }
  >
    wallet_file_import.description
  </Text>
  <View
    style={
      Object {
        "marginTop": 30,
        "width": "90%",
      }
    }
  >
    <TextInput
      autoCompleteType="password"
      defaultValue=""
      onChangeText={[Function]}
      placeholder="wallet_file_import.password_placeholder"
      placeholderTextColor="#BFBDBD"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="newPassword"
    />
  </View>
  <View
    style={
      Object {
        "bottom": 40,
        "position": "absolute",
        "width": "80%",
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2AB2E2",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
          }
        }
      >
        restore_btn
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test positive 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2AB2E2",
        "fontFamily": "Roboto-Regular",
        "fontSize": 25,
        "marginTop": 80,
      }
    }
  >
    wallet_file_import.title
  </Text>
  <Text
    style={
      Object {
        "color": "#888888",
        "fontFamily": "Roboto-Regular",
        "fontSize": 15,
        "marginTop": 40,
        "textAlign": "center",
        "width": "90%",
      }
    }
  >
    wallet_file_import.description
  </Text>
  <View
    style={
      Object {
        "marginTop": 30,
        "width": "90%",
      }
    }
  >
    <TextInput
      autoCompleteType="password"
      defaultValue=""
      onChangeText={[Function]}
      placeholder="wallet_file_import.password_placeholder"
      placeholderTextColor="#BFBDBD"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#CCCCCC",
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "height": 40,
          },
          Object {
            "width": "100%",
          },
        ]
      }
      textContentType="newPassword"
    />
  </View>
  <View
    style={
      Object {
        "bottom": 40,
        "position": "absolute",
        "width": "80%",
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2AB2E2",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Roboto-Regular",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
          }
        }
      >
        restore_btn
      </Text>
    </View>
  </View>
</View>
`;
